storage/innobase/trx/trx0trx.cc

```diff
@@ -2468,3 +2507,63 @@ trx_set_rw_mode(
 		trx->read_view.set_creator_trx_id(trx->id);
 	}
 }
+
+/*************************************************************//**
+Return difference in uncommitted count for a single undo record. */
+ib_int64_t get_diff_from_rec(
+/*============*/
+    trx_undo_rec_t* undo_rec,		/*!< in: undo record */
+    table_id_t table_id)           	/*!< in: table ID */
+{
+	ulint type, cmpl_info;
+	bool updated_extern;
+	undo_no_t undo;
+	table_id_t rec_table_id;
+
+	trx_undo_rec_get_pars(undo_rec, &type, &cmpl_info, &updated_extern, &undo,
+		&rec_table_id);
+
+	if (rec_table_id != table_id)
+		return 0;
+
+	switch (type) {
+		case TRX_UNDO_INSERT_REC:
+			return 1;
+		case TRX_UNDO_UPD_DEL_REC:
+		case TRX_UNDO_DEL_MARK_REC:
+			return -1;
+		default:
+			return 0;
+	}
+}
+
+/*************************************************************//**
+Return number of uncommitted records for table within transaction
+@param[in]	table 	table to count uncommitted records for
+*/
+ib_int64_t trx_t::uncommitted_count(dict_table_t* table)
```

Comment:
Prefer the standard type name `int64_t`.

Also, use `const` whenever possible, and do not add `/****/` or `/*====*/` comments. I would suggest something like the following (if we were to keep this kind of interface):
```c++
/** Determine the change to uncommitted records for a table.
@param table     persistent table
@return the change to uncommitted records for a table in the transaction */
int64_t trx_t::uncommitted_count(const dict_table_t& table) const
```